erpnext/buying/doctype/supplier/supplier.py

```diff
@@ -153,3 +160,22 @@ def get_supplier_primary_contact(doctype, txt, searchfield, start, page_len, fil
 		""",
 		{"supplier": supplier, "txt": "%%%s%%" % txt},
 	)
+
+
+def get_users_query():
+	user = frappe.qb.DocType("User")
+	hasRole = frappe.qb.DocType("Has Role")
+	q = (
+		frappe.qb.from_(user)
+		.inner_join(hasRole)
+		.on(user.name == hasRole.parent)
+		.select(user.name, user.full_name)
+		.where(hasRole.role == "Supplier")
+	)
+	return q
+
+
+@frappe.whitelist()
+def get_users_with_supplier_role(doctype, txt, searchfield, start, page_len, filters):
```

Comment:
- Use `txt` parameter for searching user by name.
- Use `start` and `page_len` parameter to limit results (usually 20 are returned)